src/models/transactions/offerCreate.ts

```diff
@@ -34,17 +34,10 @@ export interface OfferCreate extends BaseTransaction {
     if (tx.TakerPays === undefined)
         throw new ValidationError("OfferCreate: missing field TakerPays")
 
-    const isIssuedCurrency = (obj: IssuedCurrencyAmount): boolean => {
-        return Object.keys(obj).length === 3 
-            && typeof obj.value === 'string'
-            && typeof obj.issuer === 'string'
-            && typeof obj.currency === 'string'
-    }
-
-    if (typeof tx.TakerGets !== 'string' && !isIssuedCurrency(tx.TakerGets))
+    if (typeof tx.TakerGets !== 'string' && !isAmount(tx.TakerGets))
         throw new ValidationError("OfferCreate: invalid TakerGets")
 
-    if (typeof tx.TakerPays !== 'string' && !isIssuedCurrency(tx.TakerPays))
+    if (typeof tx.TakerPays !== 'string' && !isAmount(tx.TakerPays))
```

Comment:
These two lines can just be `!isAmount(tx.TakerGets)` instead of `typeof tx.TakerGets !== 'string' && !isAmount(tx.TakerGets)` because isAmount does the string check already